records/generator.go

```diff
@@ -105,9 +106,15 @@ func (rg *RecordGenerator) findMaster(leader string, masters []string) (state.St
 // Loads state.json from mesos master
 func (rg *RecordGenerator) loadFromMaster(ip string, port string) (sj state.State) {
 	// REFACTOR: state.json security
-	url := "http://" + ip + ":" + port + "/master/state.json"
 
-	req, err := http.NewRequest("GET", url, nil)
+	master := net.JoinHostPort(ip, port)
```

Comment:
This variable is unnecessary. You can inline it in the url struct instantiation.